packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -348,6 +348,29 @@ - (NSString *)getTemporaryFilePathWithExtension:(NSString *)extension
 - (BOOL)setCaptureSessionPreset:(FLTResolutionPreset)resolutionPreset withError:(NSError **)error {
   switch (resolutionPreset) {
     case FLTResolutionPresetMax:
+      {
+          AVCaptureDeviceFormat *bestFormat = [self getHighestResolutionFormatFor:_captureDevice];
+          if ( bestFormat ) {
+              _videoCaptureSession.sessionPreset = AVCaptureSessionPresetInputPriority;
+              if ( [_captureDevice lockForConfiguration:NULL] == YES ) {
+
+                  // set best device format and finish device configuration
+                  _captureDevice.activeFormat = bestFormat;
```

Comment:
is this only required in best format but not other resolutions?

Reply:
In all other cases device is configured with sessionPreset through capture session, like this:
 _videoCaptureSession.sessionPreset = AVCaptureSessionPreset3840x2160;
 
 The max resolution case is different because there is no predetermined option